File: examples/node/scrape_basic.js
#!/usr/bin/env node
/**
 * Basic smoke test for Node client against local self-hosted ASB API.
 */
const { AsbClient, AsbError } = require('../../clients/node');

async function main() {
  const base = process.env.ASB_BASE_URL || 'http://localhost:8000';
  const key = process.env.ASB_API_KEY;
  if (!key) {
    console.error('ERROR: Set ASB_API_KEY (from API startup logs)');
    process.exit(1);
  }

  const client = new AsbClient({ baseUrl: base, apiKey: key });

  console.log('== Health ==');
  console.log(await client.health());

  console.log('\n== Scrape ==');
  try {
    const r = await client.scrape({ url: 'https://example.com', method: 'GET', region: 'jp', timeout: 20 });
    console.log('status:', r.status);
    console.log('html_len:', (r.html || '').length);
    console.log('request_id:', r.metadata && r.metadata.request_id);
  } catch (e) {
    if (e instanceof AsbError) {
      console.error('Scrape error:', e.message, e.statusCode);
      process.exit(2);
    }
    throw e;
  }

  console.log('\n== Session create/get/delete ==');
  try {
    const s = await client.createSession({ region: 'jp' });
    console.log('created:', s.session_id);
    console.log(await client.getSession(s.session_id));
    await client.deleteSession(s.session_id);
    console.log('deleted');
  } catch (e) {
    if (e instanceof AsbError) console.log('Session flow (may fail without persistence):', e.message);
  }

  console.log('\n== Usage ==');
  try {
    console.log(await client.getUsage());
  } catch (e) {
    if (e instanceof AsbError) console.log('Usage not available:', e.message);
  }

  console.log('\n== Billing portal ==');
  try {
    console.log(await client.getBillingPortal());
  } catch (e) {
    if (e instanceof AsbError) console.log('Billing portal not available in this mode:', e.message);
  }

  console.log('\nNode examples completed.');
}

main().catch(err => { console.error(err); process.exit(1); });

File: examples/python/scrape_basic.py
#!/usr/bin/env python3
"""Basic smoke test using the Python client against local self-hosted API."""

import os
import sys

# Allow running from repo root without install
sys.path.insert(0, os.path.join(os.path.dirname(__file__), "../../clients/python"))

from asb_client import AsbClient, AsbError


def main():
    base = os.getenv("ASB_BASE_URL", "http://localhost:8000")
    key = os.getenv("ASB_API_KEY")
    if not key:
        print("ERROR: Set ASB_API_KEY (from `python -m asb_api` startup logs)", file=sys.stderr)
        sys.exit(1)

    client = AsbClient(base_url=base, api_key=key)

    print("== Health ==")
    h = client.health()
    print(h)

    print("\n== Scrape (GET example.com) ==")
    try:
        r = client.scrape(url="https://example.com", method="GET", region="jp", timeout=20)
        print("status:", r.get("status"))
        print("html_len:", len(r.get("html") or ""))
        print("request_id:", r.get("metadata", {}).get("request_id") if r.get("metadata") else None)
    except AsbError as e:
        print("Scrape failed:", e)
        sys.exit(2)

    print("\n== Create session ==")
    try:
        s = client.create_session(region="jp", fingerprint="general")
        print(s)
        sid = s["session_id"]

        print("\n== Get session ==")
        print(client.get_session(sid))

        print("\n== Delete session ==")
        client.delete_session(sid)
        print("deleted ok")
    except AsbError as e:
        print("Session ops failed (may be ok if no persistence):", e)

    print("\n== Usage ==")
    try:
        u = client.get_usage()
        print(u)
    except AsbError as e:
        print("Usage failed:", e)

    print("\n== Billing portal (optional) ==")
    try:
        p = client.get_billing_portal()
        print(p)
    except AsbError as e:
        print("Billing portal not available (expected in self-hosted without Stripe):", e)

    print("\nAll basic examples completed successfully.")


if __name__ == "__main__":
    main()
